modules/git/git_commits/exercise_slides/contents.tex

```diff
@@ -0,0 +1,120 @@
+\title{%
+  Good commits - Exercises
+}
+\author{Amanda Berg}
+\institute{%
+  KTH EECS
+}
+
+\mode<article>{\maketitle}
+\mode<presentation>{%
+  \begin{frame}
+    \maketitle
+  \end{frame}
+}
+
+\mode*
+
+
+\section{Exercise 1 - Commit messages}
+\begin{frame}
+\begin{block}{Sum over list}
+    We wish to add up all non-negative numbers in a list. We have the following piece of code:
+\end{block}
+\begin{block}{Code}
+    \lstinputlisting[]{sum_list.txt}
+\end{block}
+\end{frame}
+
+\begin{frame}
+\begin{block}{Code}
+    \lstinputlisting[]{sum_list.txt}
+\end{block}
+
+\begin{alertblock}{Note:}
+    \lstinline{for (int i = 1; i < list.length; i++)}\\
+    We will miss an element in the loop!
+\end{alertblock}
+\end{frame}
+
+
+\begin{frame}
+\begin{block}{Old}
+    \lstinputlisting[]{sum_list.txt}
+\end{block}
+
+\begin{block}{New}
+    We change row 2 to \\
+    \lstinline{for (int i = 0; i < list.length; i++)}
+\end{block}
+\end{frame}
+
+\begin{frame}
+\begin{question}
+What would be a good commit message for this change? Why? %fbf-question
+\end{question}    
+\end{frame}
+
+\begin{frame}
+\begin{block}{Examples}
+    We will now look at three examples and discuss what is good and what is bad in them
+\end{block}
+\end{frame}
+
+\subsection{Examples}
+\begin{frame}
+\begin{block}{Message 1}
+    \enquote{Fixed bug}
+\end{block}
+
+\begin{question}
+\begin{itemize}
+    \pause
+    \item What is good about this message?
+    \pause
+    \item What is bad about it?
+\end{itemize}
+\end{question}
+    
+\end{frame}
+
+\begin{frame}
+\begin{block}{Message 2}
+    \enquote{Fix indexing bug where the first element is missed every time the loop is run in the sum\_positive-function}
+\end{block}
+
+\begin{question}
+\begin{itemize}
+    \pause
+    \item What is good about this message?
+    \pause
+    \item What is bad about it?
+\end{itemize}
+\end{question}
+    
+\end{frame}
+
+
+\begin{frame}
+\begin{block}{Message 3}
+    \enquote{Fix indexing bug in sum\_positive()\\ \vspace{10 pt} 
+    This commit fixes a bug where the first element is missed when iterating over the list in sum\_positive(). The bug was because the iteration started on index 1 rather than index 0}
+\end{block}
+
+\begin{question}
+\begin{itemize}
+    \pause
+    \item What is good about this message?
+    \pause
+    \item What is bad about it?
+\end{itemize}
+\end{question}
+    
+\end{frame}
+
+% Make the students write a new (improved) commit message (fbf)
+
+\section{Exercise 2 - Commit plan}
```

Comment:
What do you want to do for the commit plan?

Give them an assignment (create two or three small shell scripts) and ask them to write a plan for the commits during the development work?

Reply:
Exactly! 

Do you have any ideas for the assignment? I found it difficult to come up with an example that is simple enough for them to understand, but still big enough to warrant multiple commits. Which is why I haven't added that assignment yet

Reply:
I added an assignment now. They should create three scripts. The last of which would require two commits. I also added a question about when it's suitable to branch. (Not obvious to branch for one-commit work, although it is useful too, but at least branch for the last one which requires two commits to be complete.)

Feel free to modify.

Reply:
That looks great! I'll try to record the exercise video today and upload that as well!